SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

```diff
@@ -0,0 +1,73 @@
+import SygicMaps
+import SygicUIKit
+
+
+public class SYMKBrowseMapViewController: UIViewController {
+    
+    public var useCompass = true
+    
+    override public func loadView() {
+        view = SYMKBrowserMapView()
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        initSygicMapsSDK()
+        
+        if let view = (view as? SYMKBrowserMapView) {
+            view.compass.isHidden = !useCompass
+        }
+    }
+    
+    // MARK: - Private Methods
+    
+    private func sygicSDKInitialized() {
+        SYOnlineSession.shared().onlineMapsEnabled = true
+        (view as! SYMKBrowserMapView).setupMapView()
+        setupViewDelegates()
+    }
+    
+    private func setupViewDelegates() {
+        guard let view = view as? SYMKBrowserMapView, let mapView = view.mapView else { return }
+        view.compass.delegate = self
+        mapView.delegate = self
+    }
+    
+}
+
+extension SYMKBrowseMapViewController: SYMapViewDelegate {
+    
+    public func mapView(_ mapView: SYMapView, didChangeCameraPosition geoCenter: SYGeoCoordinate, zoom: CGFloat, rotation: CGFloat, tilt: CGFloat) {
+        guard let view = view as? SYMKBrowserMapView else { return }
+        
+        if useCompass {
+            view.compass.isHidden = (rotation == 0)
+            view.compass.viewModel = SYUICompassViewModel(course: Double(rotation), autoHide: false)
```

Comment:
Toto by som ja asi najradsej zmenil na default mvc, teda rotation by bola normalna property, ktora by mala `didSet` a urobila by v podstate to iste.
Zalezi ale ci to vieme takto jednoducho implementovat aj v ostatnych UI prvkoch

Reply:
ten compass este neni uplne hotovy. za mna podme cestou viewmodelov. ked budeme logiku view davat do controllera, vznikne z toho bordel. v compass modeli je (zatial je to v tom view, ale presuniem to do modelu) logika na auto hide (to co si ty nastavil cez view.compass.isHidden = (rotation == 0).
za mna by to malo skor vyzerat tak, ze v setupCompass ma byt 
`compass.viewModel = SYUICompassViewModel(course: 0, autoHide: true)`

a potom v tejto metode "len" 
`        if useCompass {
            var newModel = SYUICompassViewModel(with: view.compass.viewModel)
            newModel.compassCourse = Double(rotation)
            view.compass.viewModel = newModel
        }`

Reply:
edit: opravujem, to prvotne setnutie compass modelu nema byt vo view, ale v browsemapviewcontrolleri

Reply:
ked dorobime viewmodel pre symapview, oddedime ho od SYUICompassProperties. instanciu bude drzat zrejme tento controller, potom update compasu bude zas o nieco kratsi `self.mapViewModel.compassCourse = Double(rotation) view.compass.viewModel = self.mapViewModel }`